src/autoscaler/routes/routes.go

```diff
@@ -59,88 +62,109 @@ const (
+}
 
-	instance.apiRoutes = instance.apiOpenRoutes.PathPrefix("/v1/apps").Subrouter()
-	instance.apiRoutes.Path(PublicApiScalingHistoryPath).Methods(http.MethodGet).Name(PublicApiScalingHistoryRouteName)
-	instance.apiRoutes.Path(PublicApiAggregatedMetricsHistoryPath).Methods(http.MethodGet).Name(PublicApiAggregatedMetricsHistoryRouteName)
+func (r *Router) registerSchedulerRoutes() {
+	r.router.Path(SchedulePath).Methods(http.MethodPut).Name(UpdateScheduleRouteName)
+	r.router.Path(SchedulePath).Methods(http.MethodDelete).Name(DeleteScheduleRouteName)
+}
 
-	instance.apiPolicyRoutes = instance.apiOpenRoutes.Path(PublicApiPolicyPath).Subrouter()
-	instance.apiPolicyRoutes.Path("").Methods(http.MethodGet).Name(PublicApiGetPolicyRouteName)
-	instance.apiPolicyRoutes.Path("").Methods(http.MethodPut).Name(PublicApiAttachPolicyRouteName)
-	instance.apiPolicyRoutes.Path("").Methods(http.MethodDelete).Name(PublicApiDetachPolicyRouteName)
+func (r *Router) CreateApiPublicSubrouter() *mux.Router {
+	publicApiRoutes := r.router.PathPrefix("").Subrouter()
+	publicApiRoutes.Path(PublicApiInfoPath).Methods(http.MethodGet).Name(PublicApiInfoRouteName)
+	publicApiRoutes.Path(PublicApiHealthPath).Methods(http.MethodGet).Name(PublicApiHealthRouteName)
 
-	return instance
+	return publicApiRoutes
+}
+
+func (r *Router) CreateApiSubrouter() *mux.Router {
+	apiRoutes := r.router.PathPrefix("/v1/apps").Subrouter()
+	apiRoutes.Path(PublicApiScalingHistoryPath).Methods(http.MethodGet).Name(PublicApiScalingHistoryRouteName)
+	apiRoutes.Path(PublicApiAggregatedMetricsHistoryPath).Methods(http.MethodGet).Name(PublicApiAggregatedMetricsHistoryRouteName)
+	return apiRoutes
+}
+
+func (r *Router) CreateApiPolicySubrouter() *mux.Router {
+	apiPolicyRoutes := r.router.Path(PublicApiPolicyPath).Subrouter()
+	apiPolicyRoutes.Path("").Methods(http.MethodGet).Name(PublicApiGetPolicyRouteName)
+	apiPolicyRoutes.Path("").Methods(http.MethodPut).Name(PublicApiAttachPolicyRouteName)
+	apiPolicyRoutes.Path("").Methods(http.MethodDelete).Name(PublicApiDetachPolicyRouteName)
+	return apiPolicyRoutes
+}
+
+func (r *Router) GetRouter() *mux.Router {
+	return r.router
+}
+
+var autoScalerRouteInstance = NewRouter()
+
+func init() {
+	autoScalerRouteInstance.RegisterRoutes()
 }
 
 func MetricsCollectorRoutes() *mux.Router {
-	return autoScalerRouteInstance.metricsCollectorRoutes
+	return autoScalerRouteInstance.GetRouter()
 }
 
 func EventGeneratorRoutes() *mux.Router {
-	return autoScalerRouteInstance.eventGeneratorRoutes
+	return autoScalerRouteInstance.GetRouter()
 }
 
 func ScalingEngineRoutes() *mux.Router {
-	return autoScalerRouteInstance.scalingEngineRoutes
+	return autoScalerRouteInstance.GetRouter()
 }
 
 func MetricsForwarderRoutes() *mux.Router {
-	return autoScalerRouteInstance.metricsForwarderRoutes
+	return autoScalerRouteInstance.GetRouter()
 }
 
 func SchedulerRoutes() *mux.Router {
-	return autoScalerRouteInstance.schedulerRoutes
+	return autoScalerRouteInstance.GetRouter()
```

Comment:
I'm not sure what these are for if they all return the same thing?

Reply:
I still don't get the new setup, can you explain this change?

Reply:
Explained in https://github.com/cloudfoundry/app-autoscaler-release/pull/3320#discussion_r1856908168